xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-ui/src/main/resources/Refactoring/Code/RefactoringConfiguration.xml

```diff
@@ -302,4 +338,131 @@
+        <unmodifiable>0</unmodifiable>
+        <values>long|short|default|forbid</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </cache>
+      <code>
+        <contenttype>PureText</contenttype>
+        <disabled>0</disabled>
+        <editor>PureText</editor>
+        <name>code</name>
+        <number>2</number>
+        <prettyName>Code</prettyName>
+        <rows>20</rows>
+        <size>50</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.TextAreaClass</classType>
+      </code>
+      <contentType>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <freeText>forbidden</freeText>
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>contentType</name>
+        <number>6</number>
+        <prettyName>Content Type</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator> </separator>
+        <separators>|, </separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>CSS|LESS</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </contentType>
+      <name>
+        <disabled>0</disabled>
+        <name>name</name>
+        <number>1</number>
+        <prettyName>Name</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </name>
+      <parse>
+        <disabled>0</disabled>
+        <displayFormType>select</displayFormType>
+        <displayType>yesno</displayType>
+        <name>parse</name>
+        <number>4</number>
+        <prettyName>Parse content</prettyName>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.BooleanClass</classType>
+      </parse>
+      <use>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <freeText>forbidden</freeText>
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>use</name>
+        <number>3</number>
+        <prettyName>Use this extension</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator> </separator>
+        <separators>|, </separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>currentPage|onDemand|always</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </use>
+    </class>
+    <property>
+      <cache>long</cache>
+    </property>
+    <property>
+      <code>#if (!$xwiki.hasRecycleBin())
+## Hides the section of the menu in which the refactoring is displayed.
+## This should be removed if new configuration values must be displayed regardless of the recycle bin status.
+#set ($section = $doc.getObject('XWiki.ConfigurableClass').displayInSection)
```

Comment:
Have you tested it? I'm not sure to understand the logic here. You're calling this code in all pages right now, so it means if I'm in administration accessing any page with a XWiki.ConfigurableClass, it will hides the section of this page. 
Here it doesn't apply only to the recycle bin configurable class.

Reply:
Unless I'm wrong because `$doc` only apply to  RefactoringConfiguration document, because the CSSX is used with "Always". Still sounds a bit fragile.

Reply:
I did some more test with a "no cache" policy and `$doc` always refers to `Refactoring.Code.RefactoringConfiguration`.
I can hardcore I you think it makes the code safer.

Reply:
> I can hardcore it if you think it makes the code safer.

Yeah I do think it would be safer and more readable here: you only want that specific document to be checked. Note that it has nothing to do with the cache policy here, it all depends when the UIX is executed (e.g. with what doc in the context).

Reply:
`$xwiki.getDocument('Refactoring.Code.RefactoringConfiguration')` is now affected to `$doc`.
https://github.com/xwiki/xwiki-platform/pull/1413/files#diff-3c735d937df88dc14ac06a727039ad99c06968b61044fa89899f7fb5564be338R453

Reply:
Hmmmm. Not good: `$doc` is the standard Velocity variable for the current document, here you're overriding it so if you have any velocity script executed in the same context after this one and using `$doc`, it will be a big mess. Better introduce a dedicated variable name for it. In general you don't want to override variable names in velocity...

Reply:
`$doc` replaced by `$refactoringConfigurationDoc`
https://github.com/xwiki/xwiki-platform/pull/1413/files#diff-3c735d937df88dc14ac06a727039ad99c06968b61044fa89899f7fb5564be338R453